src/identity/interaction/email-password/storage/EmailPasswordResourceStoreStorageAdapter.ts

```diff
@@ -0,0 +1,136 @@
+import assert from 'assert';
+import { hash, compare } from 'bcrypt';
+import { v4 } from 'uuid';
+import type { ResourceIdentifier } from '../../../../ldp/representation/ResourceIdentifier';
+import { trimTrailingSlashes } from '../../../../util/PathUtil';
+import type { KeyValueStore } from '../../../storage/KeyValueStore';
+import { EmailPasswordStorageAdapter } from './EmailPasswordStorageAdapter';
+
+export interface EmailPasswordResourceStoreStorageAdapterArgs {
+  baseUrl: string;
+  storagePathname: string;
+  store: KeyValueStore;
+  saltRounds: number;
+}
+
+/**
+ * A payload to persist a user account
+ */
+export interface EmailPasswordAccountPayload {
+  webId: string;
+  email: string;
+  password: string;
+}
+
+/**
+ * A payload to persist the fact that a user
+ * has requested to reset their password
+ */
+export interface EmailPasswordForgotPasswordPayload {
+  email: string;
+  recordId: string;
+}
+
+/**
+ * A EmailPasswordStorageAdapter that uses a ResoucreStore
+ * to persist its information.
+ */
+export class EmailPasswordResourceStoreStorageAdapter extends EmailPasswordStorageAdapter {
```

Comment:
```suggestion
export class EmailPasswordResourceStore extends EmailPasswordStorageAdapter {
```

Reply:
I disagree. The EmailPasswordResourceStoreStorageAdapter is not a ResourceStore. So, calling it that would be false advertising.